pkg/pushsync/pushsync.go

```diff
@@ -130,6 +143,60 @@ func (ps *PushSync) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream)
 				return fmt.Errorf("chunk store: %w", err)
 			}
 
+			// push the chunk to the closest peers in parallel for replication
+			// any errors here should NOT impact the rest of the handler
+			var skipPeers []swarm.Address
+			for i := 0; i < nPeersToPushsync; i++ {
+				peer, err := ps.closestPeer(chunk.Address(), skipPeers, true)
+				if err != nil {
+					ps.logger.Tracef("pushsync replication closest peer: %w", err)
+					break
+				}
+				skipPeers = append(skipPeers, peer)
+				go func(peer swarm.Address) {
+
+					var err error
+					defer func() {
+						if err != nil {
+							ps.logger.Tracef("pushsync replication: %v", err)
+						}
+					}()
+
+					receiptPrice := ps.pricer.PeerPrice(peer, chunk.Address())
+
```

Comment:
So, this changed recently with communication of prices in headers.
In the new system, when we send out pricing headers before the request, we guess that peers who appear to be in the same neighborhood as us will charge us 0, however, when the response headers are returned, this could prove false (if the peer advertises prices for a higher than neighborhood depth, or just simply communicates a price anyway)

When we do in-neighborhood actions for push sync, we probably want to skip the peers who communicate a non-zero price in their response headers, as nobody will cover for costs arising from this, but engage in the attempt to the extent of exchanging price related headers anyway towards the peers who appear to be within the neighborhood